subprojects/build-operations/src/main/java/org/gradle/trace/buildops/BuildOperationTrace.java

```diff
@@ -46,6 +57,51 @@ public void started(BuildOperationDescriptor buildOperationDescriptor, Operation
                 writer.finished();
             }
         }
+    }
+
+    private static class BuildOperationDurationRecordingListener extends DetachingBuildOperationListener {
+
+        private final Class<?> capturedBuildOperation;
+
+        private AsyncWriter<Long> writer;
+
+        BuildOperationDurationRecordingListener(String capturedBuildOperation, File outFile, BuildOperationListenerManager manager) throws ClassNotFoundException {
+            super(outFile, manager);
+            this.capturedBuildOperation = Class.forName(capturedBuildOperation + "$Details");
```

Comment:
Ugh...

Reply:
Would you prefer that the `$Details` is passed in?

Reply:
Not sure. It would be great to have some typed access to these things is all. Nothing to change here I guess.